### DISCUSSION TITLE
tableExport file extension and export path

### DISCUSSION THREAD

**Discussion by white3km** [yahoo/elide#2437] [Elide 5]
2 questions:

1. Are there any plans to add file type extensions to table export files for csv and json? I know you can override the default ResultsStorageEngine but I was just curious if that's the best way or if this feature is coming in the near future.
2. Our app is behind a proxy that adds a path prefix (ex. /app) so our result url doesn't quite line up with the url that an end user would need to use. Any suggestions on how we can configure the async calls to have this additional path prefix?

**Comment by aklish** [comment]
For 2, if you are using Spring, you can write your own forwarding controller to forward (not redirect) /app/export to /export.  There may also be a way to do it with a Spring WebMvcConfigurer but I'm not sure if that is only for static path routing.

For 1, we don't have plans to add any more.  To add a new format, you'll need to create a new `TableExportFormatter` and register it 
[here](https://github.com/yahoo/elide/blob/master/elide-spring/elide-spring-boot-autoconfigure/src/main/java/com/yahoo/elide/spring/config/ElideAsyncConfiguration.java#L87-L89)

**Reply by nerdstep** [reply]
@aklish To clarify @white3km's question on #1 -- we're not looking to add a new formatter, just append the file extension to the generated file, which right now is just named with the UUID (no file extension), i.e. if the specified `resultType` is `CSV`, then the expected filename would be `<id>.csv`

**Reply by aklish** [reply]
Made an issue out of it.  Seems like a reasonable thing to want to do.

**Comment by moizarafat** [comment]
Submitted https://github.com/yahoo/elide/pull/2453 to add extensions

**Comment by aklish** [comment]
Fixed in 6.0.6
